### PR TITLE
Adding webdav support

### PR DESCRIPTION
**Pull request by Sergio-Muriel**
Adding webdav support for #44

### CODE REVIEW

**Comment by lexoyo** [comment]
@Sergio-Muriel you are my star !!

remarks I have

* npm module `form-data` needs to be added to `package.json` (`npm i --save form-data`)
* I can not get it to work with [framadrive](https://framadrive.org/)'s owncloud: when `dav.list` is called it returns an HTML page and a `301 redirect` code. Not sure why, do you think it is a problem with framadrive?

File: lib/services/webdav.js [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+  var dav = webdavClientsArray[request.session.webdavClientId];
+
+  dav.list(path, function(err, data) {
+
+    path = path.replace(/\/+/g,'/'); // Remove doble / on the path
+    path = path.replace(/\/$/,'')+'/'; // Force to have last / on the path
+    if (err!==false){
+      filesData=[];
+      parseString(data, function (err, result) {
+        if(result['d:multistatus'] && result['d:multistatus']['d:response'])
```
Comment (lexoyo) [inline lib/services/webdav.js]: Handle the case where `err` is not null here
`if(!err && result['d:multistatus'] && result['d:multistatus']['d:response'])`

File: lib/services/webdav.js [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+                    bytes: size,
+                    modified: mdate,
+                    name: name,
+                    is_dir: is_dir
+                  });
+                }
+              });
+        }
+        else
+        {
```
Comment (lexoyo) [inline lib/services/webdav.js]: ... and then I would add 
```
cbk({success:false, message: data});
```

**Comment by lexoyo** [comment]
I also tried with a box.com account (davs://dav.box.com/dav as the host) but it says my credentials are invalid :(

File: lib/services/webdav.js (hunk 1 above)
Comment (Sergio-Muriel) [inline lib/services/webdav.js]: Added to PR

File: lib/services/webdav.js (hunk 2 above)
Comment (Sergio-Muriel) [inline lib/services/webdav.js]: Added to PR

**Comment by Sergio-Muriel** [comment]
For the framadrive problem, i tested on my account without problem.

- I think it could be a problem with the default port which was 80, and it is most common to have https, so i switched the default value to that.
- Fixed d:getLasModified value, that can be provided by data-attribute or by tag value

For box.com, it is working with:
host: https://dav.box.com/dav
port: 443

**Comment by lexoyo** [comment]
:+1: excellent!
Good to go for me :)

**Comment by lexoyo** [comment]
Much better @Sergio-Muriel !!

Still a few things

* the current folder is listed (its name)
* `is_dir` is always `true`
* `rm` or `cp` a non existing file returns `{"success":false,"message":false}` (wrong message)
* `mkdir` works but crashes the server

We are almost there for a merge and release !!!

**Comment by Sergio-Muriel** [comment]
Almost all fixed, but i cannot reproduce the bug:
- the current folder is listed (its name)
On which dav server does it happen?

**Comment by lexoyo** [comment]
> the current folder is listed (its name) On which dav server does it happen?

on framadrive, it looks like this:

`[{"bytes":0,"modified":"2016-02-22T17:38:50.000Z","name":"test","is_dir":true},{"bytes":4126,"modified":"2016-02-22T17:32:04.000Z","name":"github-spritesheet.png","is_dir":true}]`

where test is the current directory

**Comment by Sergio-Muriel** [comment]
fixed

**Comment by JbIPS** [comment]
I tried it this morning, great work!

I've got a list for you as well:
* Removing all the remaining `console.log`
* [This](https://github.com/Sergio-Muriel/unifile/blob/cloud/lib/services/webdav.js#L445) looks weird. `if(err)` would be better but it doesn't look like your in the case of handling the error.

I'm very excited by this new service :smile: !

**Comment by Sergio-Muriel** [comment]
Wow, that's true. the variable name was err, should be 'success'.

**Comment by lexoyo** [comment]
Nice! Are we good to go then? :+1:

**Comment by JbIPS** [comment]
One line escape the correction: https://github.com/silexlabs/unifile/pull/53/files#diff-813ebf34c67110916d54eeb39f12d07eR517

Aside from that, LGTM :+1:

**Comment by Sergio-Muriel** [comment]
done

**Pull request merged**